rasa/nlu/classifiers/logistic_regression_classifier.py

```diff
@@ -0,0 +1,190 @@
+            # for intent classification we're only interested in sentence features
+            sparse_feats = e.get_sparse_features(attribute=TEXT)[1]
+            dense_feats = e.get_dense_features(attribute=TEXT)[1]
+            together = hstack(
+                [
+                    csr_matrix(sparse_feats.features if sparse_feats else []),
+                    csr_matrix(dense_feats.features if dense_feats else []),
+                ]
+            )
+            X.append(together)
+            y.append(e.get(INTENT))
+        return vstack(X), y
+
+    def train(self, training_data: TrainingData) -> Resource:
+        """Train the intent classifier on a data set."""
+        X, y = self._create_training_matrix(training_data)
+        if X.shape[0] == 0:
+            logger.debug(
+                f"Cannot train '{self.__class__.__name__}'. No data was provided. "
+                f"Skipping training of the classifier."
+            )
+            return self._resource
+
+        self.clf.fit(X, y)
+        self.persist()
+
+        return self._resource
+
+    @classmethod
+    def create(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> GraphComponent:
+        """Creates a new untrained component (see parent class for full docstring)."""
+        return cls(config, execution_context.node_name, model_storage, resource)
+
+    def process(self, messages: List[Message]) -> List[Message]:
+        """Return the most likely intent and its probability for a message."""
+        X = self._create_X(messages)
+        pred = self.clf.predict(X)
+        probas = self.clf.predict_proba(X)
+        for idx, message in enumerate(messages):
+            intent = {"name": pred[idx], "confidence": probas[idx].max()}
+            intents = self.clf.classes_
+            intent_info = {
+                k: v
+                for i, (k, v) in enumerate(zip(intents, probas[idx]))
+                if i < LABEL_RANKING_LENGTH
+            }
+            intent_ranking = [
+                {"name": k, "confidence": v} for k, v in intent_info.items()
+            ]
+            message.set("intent", intent, add_to_output=True)
+            message.set("intent_ranking", intent_ranking, add_to_output=True)
+        return messages
+
+    def persist(self) -> None:
+        """Persist this model into the passed directory."""
+        with self._model_storage.write_to(self._resource) as model_dir:
+            dump(self.clf, model_dir / f"{self.name}.joblib")
+
+    @classmethod
+    def load(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> GraphComponent:
+        """Loads trained component (see parent class for full docstring)."""
+        with model_storage.read_from(resource) as model_dir:
```

Comment:
To cover the case where there is no persisted data for the component, this should be covered in an exception block catching `ValueError`.

Reply:
Is there an example of this? When would there not be any persisted data?

Reply:
[This](https://github.com/RasaHQ/rasa/blob/main/rasa/nlu/classifiers/diet_classifier.py#L1074) would be an example. I don't think this would ever happen programatically, but I think if e.g. someone deleted the individual cached model folder in `.rasa`, but not the database containing the pointers to the cached components, the lookup would fail.

Reply:
This begs the question: do we have a unit test for this behavior?